lib/rspec/core/world.rb

```diff
@@ -128,11 +128,18 @@ def reporter
       end
 
       # @private
-      def source_cache
-        @source_cache ||= begin
-          RSpec::Support.require_rspec_core "source"
-          Source::Cache.new(@configuration)
+      def source_from_file(path)
+        unless instance_variable_defined?(:@sources_by_path)
+          RSpec::Support.require_rspec_support 'source'
+          @sources_by_path = {}
```

Comment:
Do we need to clear `@sources_by_path` (e.g. `@sources_by_path.clear`) when `world.reset` is called?  I'm thinking maybe we do....